src/vs/workbench/parts/extensions/browser/extensionsActions.ts

```diff
@@ -1717,6 +1788,149 @@ export class DisabledStatusLabelAction extends Action {
 	}
 }
 
+// TODO: test when workspace is folder
+//				TODO: test when recomendation file already exists
+//							TODO: test when current extension is already recommended
+//							TODO: test when current extension is NOT recommended
+//				TODO: test when recomendation file doesnt exists
+// TODO: test when workspace is workspace
+//				TODO: test when recomendation file already exists
+//							TODO: test when current extension is already recommended
+//							TODO: test when current extension is NOT recommended
+//				TODO: test when recomendation file doesnt exists
+// TODO: test when workspace is empty
+// TODO: make sure we preserve comments
+
+export class RecommendToFolderAction extends AbstractConfigureRecommendedExtensionsAction {
+	private static readonly Class = 'extension-action recommend-to-folder';
+
+	private _extension: IExtension;
+	get extension(): IExtension { return this._extension; }
+	set extension(extension: IExtension) { this._extension = extension; this.update(); }
+
+	private disposables: IDisposable[] = [];
+
+	private _configurationFile: URI;
+	get configurationFile(): URI {
+		if (!this._configurationFile) {
+			this._configurationFile = this.contextService.getWorkspace().folders[0].toResource(paths.join('.vscode', 'extensions.json'));
+		}
+		return this._configurationFile;
+	}
+
+	constructor(
+		@IFileService fileService: IFileService,
+		@IWorkspaceContextService contextService: IWorkspaceContextService,
+		@IWorkbenchEditorService editorService: IWorkbenchEditorService,
+		@IJSONEditingService jsonEditingService: IJSONEditingService,
+		@ITextModelService textModelResolverService: ITextModelService,
+		@IExtensionsWorkbenchService private extensionsWorkbenchService: IExtensionsWorkbenchService
+	) {
+		super(
+			'extensions.install',
+			localize('recommendToFolder', "Recommend To Folder"),
+			`${RecommendToFolderAction.Class}`,
+			false,
+			contextService,
+			fileService,
+			editorService,
+			jsonEditingService,
+			textModelResolverService
+	);
+		this.disposables.push(this.extensionsWorkbenchService.onChange(() => this.update()));
```

Comment:
The extensionWorkbenchService change event is fired when an extension is installed/updated/uninstalled which shouldnt affect the recommend actions.

Instead, we should update on `this.contextService.onDidChangeWorkbenchState` which gets fired when workspace changes from multi-root to single root and vice versa